src/cascadia/TerminalApp/App.cpp

```diff
@@ -3,667 +3,60 @@
-                                    HRESULT settingsLoadedResult)
+    AppLogic App::Logic()
     {
-        auto title = GetLibraryResourceString(titleKey);
-        auto buttonText = RS_(L"Ok");
-
-        Controls::TextBlock warningsTextBlock;
-        // Make sure you can copy-paste
-        warningsTextBlock.IsTextSelectionEnabled(true);
-        // Make sure the lines of text wrap
-        warningsTextBlock.TextWrapping(TextWrapping::Wrap);
-
-        winrt::Windows::UI::Xaml::Documents::Run errorRun;
-        const auto errorLabel = GetLibraryResourceString(contentKey);
-        errorRun.Text(errorLabel);
-        warningsTextBlock.Inlines().Append(errorRun);
-
-        if (FAILED(settingsLoadedResult))
-        {
-            if (!_settingsLoadExceptionText.empty())
-            {
-                warningsTextBlock.Inlines().Append(_BuildErrorRun(_settingsLoadExceptionText, Resources()));
-            }
-        }
-
-        // Add a note that we're using the default settings in this case.
-        winrt::Windows::UI::Xaml::Documents::Run usingDefaultsRun;
-        const auto usingDefaultsText = RS_(L"UsingDefaultSettingsText");
-        usingDefaultsRun.Text(usingDefaultsText);
-        warningsTextBlock.Inlines().Append(usingDefaultsRun);
-
-        Controls::ContentDialog dialog;
-        dialog.Title(winrt::box_value(title));
-        dialog.Content(winrt::box_value(warningsTextBlock));
-        dialog.CloseButtonText(buttonText);
-
-        _ShowDialog(nullptr, dialog);
+        static AppLogic logic;
+        return logic;
     }
 
-    // Method Description:
-    // - Displays a dialog for warnings found while loading or validating the
-    //   settings. Displays messages for whatever warnings were found while
-    //   validating the settings.
-    // - Only one dialog can be visible at a time. If another dialog is visible
-    //   when this is called, nothing happens. See _ShowDialog for details
-    void App::_ShowLoadWarningsDialog()
+    /// <summary>
+    /// Invoked when the application is launched normally by the end user.  Other entry points
+    /// will be used such as when the application is launched to open a specific file.
+    /// </summary>
+    /// <param name="e">Details about the launch request and process.</param>
+    void App::OnLaunched(LaunchActivatedEventArgs const& /*e*/)
     {
-        auto title = RS_(L"SettingsValidateErrorTitle");
-        auto buttonText = RS_(L"Ok");
-
-        Controls::TextBlock warningsTextBlock;
-        // Make sure you can copy-paste
-        warningsTextBlock.IsTextSelectionEnabled(true);
-        // Make sure the lines of text wrap
-        warningsTextBlock.TextWrapping(TextWrapping::Wrap);
-
-        const auto& warnings = _settings->GetWarnings();
-        for (const auto& warning : warnings)
+        // if this is a UWP... it means its our problem to hook up the content to the window here.
+        if (_isUwp)

```

Comment:
I'm assuming this path isn't supposed to work right now, correct? It looks to me like at the end of this block, the `logic` is going to have it's refcount decremented and it'll get cleaned up. Seems like the App would need to hang onto it as a member... but I could be wrong.

Reply:
It doesn't work right now in that there's no Universal head showing in the project. But this code does totally work correctly if you invoke from a Universal context.

It's not going to be cleaned up. It has an outstanding reference in the static right now due to the "magic static" as far as I know after the first call and won't be cleaned, just wait for future use.